sycl/include/CL/sycl/device.hpp

```diff
@@ -188,9 +188,10 @@ class __SYCL_EXPORT device {
   /// \return true if the SYCL device has the given feature.
   bool has(aspect Aspect) const;
 
+  device(shared_ptr_class<detail::device_impl> impl) : impl(impl) {}
+
```

Comment:
Why did you need to make this constructor public?

Reply:
We have decided to undo this change and declare the function "make_device()" as a friend of the device class so that it can access this private constructor.

Reply:
same small nit here .. the location of the `impl` member got moved.  Unless there is a specific reason to move it, you should put it back where it was before.  In general, you don't want a PR to contain unneeded code changes.